### PR TITLE
cli(metadata): add diff command and dry-run flag

### LINKED ISSUE #3126: add a diff command to diff metadata on file and on server
Add a new command to the CLI which can be used to diff the metadata on the server with one that is available locally.

Usage:
```
# diff server and migrataions/metadata.yaml
hasura metadata diff

# diff server and <file>
hasura metadata diff --file <file>

# diff <file1> and <file2>
hasura metadata diff --file <file1> --file <file2>
```

`--file` flag should have a shorthand of `-f`

Here is a gist for implementing diff:
https://gist.github.com/shahidhk/f39bf76e5fdf1e7a364638de1d12114b

### PR DESCRIPTION
**Pull request by jeffsieu**
### Description

Adds a `metadata diff` command to show comparisons between two different sets of Hasura metadata.
```
# Show changes between server metadata and the exported metadata file:
hasura metadata diff

# Show changes between server metadata and that in local_metadata.yaml:
hasura metadata diff local_metadata.yaml

# Show changes between metadata from metadata.yaml and metadata_old.yaml:
hasura metadata diff metadata.yaml metadata_old.yaml
```

Also adds a `--dry-run` flag to `metadata apply` command which will print the diff and exit rather than actually applying the metadata.

### Affected components 

- CLI
- Docs

### Related Issues

#3126 

### Solution and Design

- Added `metadata_diff.go` and `metadata_diff_test.go`

### Steps to test and verify

```
hasura metadata export
# Make changes to migrations/metadata.yaml
hasura metadata diff
```

### Limitations, known bugs & workarounds

This is just a general-purpose diff. 

A more contextual diff with the understanding of metadata can be added once #3072  is merged.

### CODE REVIEW

**Comment by jeffsieu** [comment]
@shahidhk I'm not too sure why the CI tests for test_cli_with_latest_release fails...

**Comment by jeffsieu** [comment]
Oh, it was because I accidentally updated the Gopkg files to download dependencies of different versions...

**Comment by shahidhk** [comment]
Thanks @jeffsieu! I'll get back on the review today.

**Comment by jeffsieu** [comment]
@shahidhk Any updates? Or things I need to change for this PR?

**Comment by shahidhk** [comment]
All good @jeffsieu, I am just making some modifications from my end :) Thanks

**Comment by jeffsieu** [comment]
@shahidhk Alright, cheers!

File: cli/commands/metadata_diff.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	if err != nil {
+		return errors.Wrap(err, "cannot read file")
+	}
+
+	// return diffYaml(oldYaml, newYaml, o.output)
+	printDiff(string(oldYaml), string(newYaml), o.output)
+	return nil
+}
+
+func yamlToMap(y []byte) (map[string]string, error) {
```
Comment (arvi3411301) [inline cli/commands/metadata_diff.go]: Can we remove `yamlToMap` and `diffYaml` functions?

**Pull request merged**